controllers/capabilities/product_controller.go

```diff
@@ -329,3 +355,30 @@ func computeBackendUsageList(list []capabilitiesv1beta1.Backend, backendUsageMap
 
 	return result
 }
+
+func (r *ProductReconciler) removeProduct(productResource *capabilitiesv1beta1.Product) error {
+	providerAccount, err := controllerhelper.LookupProviderAccount(r.Client(), productResource.Namespace, productResource.Spec.ProviderAccountRef, r.Logger())
+	if err != nil {
+		return err
+	}
+
+	threescaleAPIClient, err := controllerhelper.PortaClient(providerAccount)
+	if err != nil {
+		return err
+	}
+
+	err = threescaleAPIClient.DeleteProduct(*productResource.Status.ID)
```

Comment:
Is there a use case where the user can delete a product CR and the status ID is `nil`? I would say it could happen, so I would add a guard at the beginning, if the ID is nil, just return.

Reply:
Same comment applies to the backend CR